impl/src/main/java/com/sun/faces/facelets/util/Classpath.java

```diff
@@ -93,18 +94,18 @@ public static URL[] search(ClassLoader cl, String prefix, String suffix, SearchA
                 if (jarFile != null) {
                     searchJar(cl, all, jarFile, prefix, suffix, advice);
                 } else {
-                    boolean searchDone = searchDir(all, new File(URLDecoder.decode(url.getFile(), "UTF-8")), suffix);
+                    boolean searchDone = searchDir(all, new File(URLDecoder.decode(url.getFile(), StandardCharsets.UTF_8)), suffix);
                     if (!searchDone) {
                         searchFromURL(all, prefix, suffix, url);
                     }
                 }
             }
         }
-        URL[] urlArray = (URL[]) all.toArray(new URL[all.size()]);
+        URL[] urlArray = all.toArray(new URL[0]);
```

Comment:
Why was `all.size()` changed by `0`? This will now unnecessarily create an empty URL array.